backend/src/test/java/com/github/redstylzz/backend/service/PaymentServiceTest.java

```diff
@@ -46,32 +54,38 @@ void shouldThrowExceptionIfCategoryDoesNotExist_OnAdd(){
 
     @Test
     void shouldReturnListAfterAddingPayment_OnAdd() {
-        Payment payment = TestDataProvider.testPayment();
-        String userID = payment.getUserID();
+        String randomUUID = UUID_STRING;
+        Payment actualPayment = TestDataProvider.testPayment();
+        actualPayment.setPaymentID(randomUUID);
+        String userID = actualPayment.getUserID();
         when(categoryRepository.existsByUserIDAndCategoryID(anyString(), anyString())).thenReturn(true);
-        when(paymentRepo.getAllByUserIDAndCategoryID(anyString(), anyString())).thenReturn(List.of(payment));
+        when(paymentRepo.getAllByUserIDAndCategoryID(anyString(), anyString())).thenReturn(List.of(actualPayment));
 
-        assertEquals(List.of(payment), underTest.addPayment(userID, payment));
+        assertEquals(List.of(Payment.convertPaymentToDTO(actualPayment)), underTest.addPayment(userID, actualPayment));
         verify(paymentRepo).save(any(Payment.class));
     }
 
     @Test
     void shouldThrowExceptionIfCategoryDoesNotExist_OnDelete() {
-        Payment payment = TestDataProvider.testPayment();
-        String userID = payment.getUserID();
+        Payment actualPayment = TestDataProvider.testPayment();
+        String userID = actualPayment.getUserID();
+        String categoryID = actualPayment.getCategoryID();
+        String paymentID = actualPayment.getPaymentID();
         when(categoryRepository.existsByUserIDAndCategoryID(anyString(), anyString())).thenReturn(false);
 
-        assertThrows(CategoryDoesNotExistException.class, () -> underTest.deletePayment(userID, payment));
+        assertThrows(CategoryDoesNotExistException.class, () -> underTest.deletePayment(userID, categoryID, paymentID));
     }
 
     @Test
     void shouldReturnListAfterDeletingPayment_OnDelete() {
-        Payment payment = TestDataProvider.testPayment();
-        String userID = payment.getUserID();
+        Payment actualPayment = TestDataProvider.testPayment();
+        String userID = actualPayment.getUserID();
+        String categoryID = actualPayment.getCategoryID();
+        String paymentID = actualPayment.getPaymentID();
         when(categoryRepository.existsByUserIDAndCategoryID(anyString(), anyString())).thenReturn(true);
-        when(paymentRepo.getAllByUserIDAndCategoryID(anyString(), anyString())).thenReturn(List.of(payment));
+        when(paymentRepo.getAllByUserIDAndCategoryID(anyString(), anyString())).thenReturn(List.of(actualPayment));
 
-        assertEquals(List.of(payment), underTest.deletePayment(userID, payment));
+        assertEquals(List.of(Payment.convertPaymentToDTO(actualPayment)), underTest.deletePayment(userID, categoryID, paymentID));
```

Comment:
assert statements should ideally never include method calls, if you need to retrieve specific values, do so beforehand and save them to a new variable